### DISCUSSION TITLE
NVIDIA RTX IO

### DISCUSSION THREAD

**Discussion by robertosfield** [vsg-dev/VulkanSceneGraph#629] [General]
NVidia have announced RTX IO for transferring data to GPU memory and uncompressing on the GPU, seems like an interesting development for anyone who wants to rapidly stream data to the GPU.

https://developer.nvidia.com/rtx-io

Requires two new extensions [VK_NV_memory_decompression and VK_NV_copy_memory_indirect,](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#memory-decompression) which are part of [Vulkan1-.3.236](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html).

I haven't looked into the details of the format supported or how portable it'll be, what fallback we could provide in case extension support is not available.  The above article mentions being open source but haven't spotted a link to any published source yet.

One of the possibilities with being able to uncompress general data on the GPU, not mentioned in the above article is storing data compressed on the GPU and just uncompressing when required.  For instance could one handle a really large CAD or Lidar dataset by having it all resident on the GPU and just uncompressing the parts that are relevant to the current view?

**Comment by Modenbach** [comment]
Hi Robert,
sounds interesting. Also of interest are unbound buffers. Does vsg support this?
Werner

**Comment by robertosfield** [comment]
On Mon, 5 Dec 2022 at 17:03, Werner Modenbach ***@***.***>
wrote:

> sounds interesting. Also of interest are unbound buffers. Does vsg support
> this?


I am not familiar with unbounded buffers.  Could you provide a link to
specs/articles discussing it?
